app/Http/Requests/ItemImportRequest.php

```diff
@@ -38,9 +40,8 @@ public function import(Import $import)
 
         $filename = config('app.private_uploads').'/imports/'.$import->file_path;
         $import->import_type = $this->input('import-type');
-        $class = ucfirst($import->import_type);
-        $classString = "App\\Importer\\{$class}Importer";
-        $importer = new $classString($filename);
+        $importer = Factory::make($filename, $this->input('import-type'));
```

Comment:
I like this, but the name `Factory` here is confusing since that's usually used in the context of testing. Additionally, I feel like, based on how simple the Factory is, it could be just a static method on the `Importer` class - but we can't call it `factory()` since that's actually used in testing. Maybe something like `Importer::makeImporter()`? But that sounds janky. `Importer::makeSpecifcImporter`? I dunno. But something along those lines should probably work better.